### ISSUE TITLE
SPMC v17.4a9: LAN access from macOS server very slow

### ISSUE THREAD

**Issue by Michelasso** [koying/SPMC#1026]
This is actually an issue I seem to have with any SSH client on my Bravia with Android 7.0, but since you guys are wizards I'll give it a shot in here. :)

This my setup:
- SPMC v17.4a9 on Sony Bravia 49XD8099 with Android 7.0 connected via Ethernet (100Mbps max)
- MacBook Pro (SSH server) with macOS High Sierra beta (but it was the same with Sierra) connected via WiFi at around 264Mbps

Basically if I use SSH to define a library of UHD (4K) videos the maximum bandwidth I get is about 25-30Mbps. While via DLNA I do get up to 90-100Mbps.

Now I don't know if it is macOS, Android TV or else. But if you know of any configuration option or even better you can make a possible fix I would be grateful.

**Comment by classicjazz** [comment]
I did a bunch of performance testing earlier this summer with earlier builds of SMPC17.

There's a Wiki article here that covers the recommended settings:
https://github.com/koying/SPMC/wiki/Recommended-Network-Settings-for-Media-File-Serving-from-a-Network-Share

To expand upon my testing, I wanted something that performed well in terms of throughput, is widely supported across OSes and devices, supported encryption, and user-based encryption (which could be limited to a SPMC-specific user+password). 

TL;DR: use SMB3 preferably, or SMB2 as a fallback.

* SSH (SFTP) was relatively slow due to the encryption overhead, as you can see
* AFP performs well and supports user authentication but is deprecated by Apple and not supported on Windows
* UPNP/DLNA doesn't always support user authentication and implementations vary widely leading to unexpected compatibility issues
* WebDAV doesn't receive as much developer love within Kodi. On a Synology NAS, you need to install a plugin. In theory it should be low overhead. And it supports user authentication. But I found dropped frames in early SMPC17 builds. SMBv3 performs better as implemented on client and server.
* SMB1 has security issues and is considerably slower than SMBv2 or 3
* NFS isn't supported by default in Windows. It uses IP-based restrictions, not user authentication. Although many benchmarks from years ago cite a speed advantage, that was with smaller files. And SAMBAv4 now performs better.

You should also see the advanced settings recommendations for caching in the Wiki.

Finally, serving large videos via WiFi will be more prone to network interference and therefore inconsistent throughput. It is possible (I stream MPEG2 video for TV via WIFi). But, I recommend a Wireless AC Wave 2 network on 5Ghz with no other active WiFi devices using your chosen channel.

**Comment by Michelasso** [comment]
@classicjazz thanks for your answer! Unfortunately using a macOS server my choices are restricted to 3, and 2 have issues. Please keep in mind that  I have to convert DTS (and anything above DD+) to AC3 due to the broken DTS passthrough:

- SSH: as we have seen it can get quite slow
- NFS: It works (using the parameter -N in nfsd), it plays some 4K movies fine but others have the audio stuttering. The maximum bandwidth I measures has been 5-6MB/s (40-48Mbps)
- DLNA: I use Plex Server as DLNA server. This is the fastest solution. Never an hiccup. But it doesn't allow to build Kodi libraries

Unless there is a way to make SAMBA working in macOS. I have never been able to share folders to devices different than Windows.

**Comment by Michelasso** [comment]
Ok, so I have (finally!!) managed to see my macOS SMB shares from SPMC/Kodi. Apparently I first had to define a smb://192.168.1.2 (my server) network resource and then I could add libraries from there. I believe macOS uses SMB3, because in the man pages it says that we can configure SMB1, SMB2 and SMB3.

The performances are exactly the same of SSH and NFS (which apparently now even slowed down): 3.5-4 MB/s max.

DLNA (from Plex server) instead reaches 11-12MB/s with 4K movies. I also tried with the TV using WiFi ac (thus 5GHz) instead of Ethernet, and with DLNA it was even slightly higher. The Mac is always using WiFi ac, since in other tests I made in the past using it wired never made any difference.

Now I understand this can be due to the awful implementation of Android (by Mediatek) in Sony Bravia, still DLNA shows that potentially the minimum required bandwidth (90-100Mbps) is there. Is there any way I can improve it with SMB or DNS?

PS: I modified the title of the issue, since I have got the problem with most LAN protocols

**Comment by Michelasso** [comment]
Ok, I have found [this app](https://itunes.apple.com/us/app/webdavnav-server/id747482894?mt=12) to use WebDAV and with it I get full bandwidth. It also builds the video libraries, which is what Kodi via DLNA is unable to do. So that would do for now. As long as anything does the job. It's too much of a nightmare to investigate why neither SMB nor NFS get a reasonable bandwidth.
